### PR TITLE
(feat) Added pagination data into getAllThreads and searchThreads

### PR DESCRIPTION
**Pull request by sarunnut1236**
- fix proto
- fix models
- fix logic

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -3,9 +3,6 @@ name: Prettier
 on:
   push:
     branches: ["**"]
-  # pull_request:
```
Comment (ThanatWongsamut) [inline .github/workflows/lint.yml:6]: What is this??

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -27,3 +24,20 @@ jobs:
 
       - name: Run Prettier check
         run: npx prettier --write . --plugin=@trivago/prettier-plugin-sort-imports
+
```
Comment (ThanatWongsamut) [inline .github/workflows/lint.yml]: I think this can be done with husky or some git hook on local, don't need to be in pipeline

**Review by ThanatWongsamut — COMMENTED** [review]
Left some comment

File: .github/workflows/lint.yml (hunk 2 above)
Comment (sarunnut1236) [reply .github/workflows/lint.yml]: deleted the worflow

**Review by ThanatWongsamut — APPROVED** [review]
LGTM

**Pull request merged**